src/AspNet.Security.OAuth.Salesforce/SalesforceAuthenticationOptions.cs

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace AspNet.Security.OAuth.Salesforce
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="SalesforceAuthenticationHandler"/>.
+    /// </summary>
+    public class SalesforceAuthenticationOptions : OAuthOptions
+    {
+        public SalesforceAuthenticationOptions OverrideHostInEndpoints(string newHost)
```

Comment:
No, it's fine. But I wonder if an `Environment` enum property wouldn't be clearer. We already use a similar pattern in the BattleNet provider for regions:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.BattleNet/BattleNetAuthenticationOptions.cs#L30